Fix spacing in "thank you" banner

sergiou87 opened the pull request:
Closes https://github.com/desktop/desktop/issues/20402

## Description

This PR wraps the banner content inside a `<span>` element, to prevent the `display: flex` attribute of the `contents` container to break spacing.

## Release notes

Notes: [Fixed] Fix spacing of text in "Thank You" banner

sergiou87 committed:
Fix rendering structure in OpenThankYouCard

Wrapped the content of the OpenThankYouCard component in a <span> element to address potential rendering or event handling issues. This ensures proper structure and behavior for the banner's interactive elements.

niik reviewed (changes requested):
It feels odd to have a span around block elements. Could we make the span a form instead? Feels like that would be. a better match considering we want to subscribe to the submit event. Additionally it feels like we should call preventDefault on the submit event just to make sure we don't run into trouble down the line with the submit event bubbling to another form.

sergiou87 commented:
Great point!!

sergiou87 committed:
Fix semantic HTML in OpenThankYouCard component

Replaced a <span> element with a <form> element to properly handle the onSubmit event in the OpenThankYouCard component. This ensures better semantic HTML and improves accessibility.

sergiou87 commented:
Fixed in 961fe35a14c7e0d00eff51ced50ae2702d3e15ce
